intake_bluesky/core.py

```diff
@@ -15,10 +15,108 @@
 import numpy
 import warnings
 import xarray
+import heapq
+from functools import wraps
 
+def to_event_pages(get_event_cursor):
+    """
+    Decorator that changes get_event_cursor which yields events,
+    to a generator function which yields event_pages.
+
+    Parameters
+    ----------
+    get_event_cursor : function
+
+    Returns
+    -------
+    get_event_pages : function
+    """
+    def get_event_pages(*args, **kwargs):
+        event_cursor = get_event_cursor(*args, **kwargs)
+        while True:
+            result = list(itertools.islice(event_cursor, 2500))
```

Comment:
Can `2500` be exposed as a configurable parameter to `to_event_pages`? Also, is the `list()` call necessary? If not, we can save on one iteration.

Reply:
updated